publisher/publisher.go

```diff
@@ -15,15 +17,24 @@ type Publisher struct {
 	specPath    string
 	commandSpec *util.CommandSpec
 	sdAPI       api.API
+	tag         string
 }
 
 // Run is a method to publish sdapi and sdstore.
 func (p *Publisher) Run() error {
 	specResponse, err := p.sdAPI.PostCommand(p.commandSpec)
 	if err != nil {
-		return fmt.Errorf("Post failed:%v", err)
+		return fmt.Errorf("Post failed: %v", err)
 	}
 
+	commandFullName := path.Join(specResponse.Namespace, specResponse.Name)
+	promoter, err := promoter.New(p.sdAPI, []string{commandFullName, specResponse.Version, p.tag})
+	if err != nil {
+		return err
+	}
+
+	promoter.Run()
```

Comment:
I think it will be better to this promoter code to separate another function and just call the function here. That make more easier to read code.

Reply:
fixed!